File: app/src/main/java/org/kikermo/thingsaudioreceiver/model/ReceiverRepository.java
package org.kikermo.thingsaudioreceiver.model;


import org.kikermo.thingsaudioreceiver.model.data.PlayPosition;
import org.kikermo.thingsaudioreceiver.model.data.PlayState;
import org.kikermo.thingsaudioreceiver.model.data.Track;

import io.reactivex.Observable;


/**
 * Created by EnriqueR on 19/02/2017.
 */

public interface ReceiverRepository {

    Observable<Track> subscribeToTrackUpdates();

    Observable<PlayState> subscribeToPlayState();

    Observable<PlayPosition> subscribeToPlayPosition();
}

File: core_api/src/main/java/org/kikermo/thingsaudio/core/api/ReceiverRepository.kt
package org.kikermo.thingsaudio.core.api

import io.reactivex.Observable
import org.kikermo.thingsaudio.core.model.PlayState
import org.kikermo.thingsaudio.core.model.Track

interface ReceiverRepository {

    fun getTrackUpdates(): Observable<Track>

    fun getPlayStateUpdates(): Observable<PlayState>

    fun getPlayPositionUpdates(): Observable<Int>
}
